File: lib/glg/SimpleViewer/MultiThreaded/Java/GlgTagRecord.java
import com.genlogic.*;

///////////////////////////////////////////////////////////////////////
// Stores information for a given GLG tag object and can be extended 
// by the application as needed.
//////////////////////////////////////////////////////////////////////
public class GlgTagRecord 
{
   GlgObject tag_obj; 
   int data_type;
   String tag_source;
   boolean if_changed;

   public GlgTagRecord() {}
}

File: lib/glg/SCADA_Viewer/CPP/GlgTagRecord.cpp

#include "GlgTagRecord.h"

GlgTagRecord::GlgTagRecord( void )
{
   tag_source = NULL;
}

GlgTagRecord::~GlgTagRecord( void )
{
   GlgFree( (void*) tag_source ); // it checks for NULL automatically
}
